matchzoo/models/arcii_model.py

```diff
@@ -0,0 +1,124 @@
+"""An implementation of ArcII Model."""
+from matchzoo import engine
+from matchzoo import layers
+import typing
+import numpy as np
+from keras.models import Model
+from keras.layers import Input, Embedding, Conv1D, Conv2D, MaxPooling2D, \
+    Dropout, Flatten
+
+
+class ArcIIModel(engine.BaseModel):
+    """
+    ArcII Model.
+
+    Examples:
+        >>> model = ArcIIModel()
+        >>> model.guess_and_fill_missing_params()
+        >>> model.build()
+
+    """
+
+    @classmethod
+    def get_default_params(cls) -> engine.ParamTable:
+        """:return: model default parameters."""
+        params = super().get_default_params()
+        params['optimizer'] = 'adam'
+        params['input_shapes'] = [(32,), (32,)]
+        params.add(engine.Param('trainable_embedding', False))
+        params.add(engine.Param('embedding_dim', 300))
+        params.add(engine.Param('vocab_size', 100))
+        params.add(engine.Param('num_blocks', 1))
+        params.add(engine.Param('kernel_1d_count', 32))
+        params.add(engine.Param('kernel_1d_size', 3))
+        params.add(engine.Param('kernel_2d_count', [32]))
+        params.add(engine.Param('kernel_2d_size', [[3, 3]]))
+        params.add(engine.Param('activation', 'relu'))
+        params.add(engine.Param('pool_2d_size', [[2, 2]]))
+        params.add(engine.Param('padding', 'same'))
+        params.add(engine.Param('dropout_rate', 0.0))
+        params.add(engine.Param('embedding_mat', None))
+        params.add(engine.Param('embedding_random_scale', 0.2))
+        return params
+
+    def _conv_pool_block(self, input: typing.Any, kernel_count: int,
+                         kernel_size: int, padding: str, activation: str,
+                         pool_size: int) -> typing.Any:
+        output = Conv2D(kernel_count,
+                        kernel_size,
+                        padding=padding,
+                        activation=activation)(input)
+        output = MaxPooling2D(pool_size=pool_size)(output)
+        return output
+
+    @property
+    def embedding_mat(self) -> np.ndarray:
+        """Get pretrained embedding for ArcII model."""
+        # Check if provided embedding matrix
+        if self._params['embedding_mat'] is None:
+            s = self._params['embedding_random_scale']
+            self._params['embedding_mat'] = \
+                np.random.uniform(-s, s, (self._params['vocab_size'],
+                                          self._params['embedding_dim']))
+        return self._params['embedding_mat']
```

Comment:
Things in `self._params` should not be altered here, or it will cause problems when saving and loading. Store the embedding matrix in `self._embedding_matrix` or something.